xclim/indices/stats.py

```diff
@@ -153,6 +155,62 @@ def fit(
     return out
 
 
+def fit_group(
+    da: xr.DataArray,
+    dist: str = "gamma",
+    freq: str = "MS",
+    window: int = 1,
+) -> xr.DataArray:
+    """Fit a resampled array averaged over a rolling window to a univariate distribution along the time dimension.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+      Input data.
+    dist : str
+      Name of the univariate distribution, such as `beta`, `expon`, `genextreme`, `gamma`, `gumbel_r`, `lognorm`, `norm`
+      (see scipy.stats).
+    freq : str
+      Resampling frequency.
+    window : int
+      Averaging window length (In units of the resampling frequency period).
+
+
+    Returns
+    -------
+    xr.DataArray
+      An array of fitted distribution parameters for each time segment
+
+
+    Notes
+    -----
+    The resampling frequency time delta must be an integer number of months. It determines time segments used to fit
+    the array group by group. For example, choosing `freq = 'MS` will give 12 time segment, i.e. months.
+    A time series for each month is used to obtain a set of fitting parameters.
+    """
+    # Freq must have a time delta which is an integer number of month
+    dts = pd.date_range("1950-01-01", periods=2, freq=freq).to_pydatetime()
+    if dts[-1].day - dts[0].day != 0 or dts[-1].second - dts[0].second != 0:
+        raise NotImplementedError(f"Resampling frequency `{freq}` not supported.")
```

Comment:
The frequency strings are limited, I believe you can hardcode this:
```suggestion
    _, base, _,  _ = parse_offset(freq)
    if base not in ['Y', 'A', 'Q', 'M']:
        raise NotImplementedError(f"Resampling frequency `{freq}` not supported.")
```
This needs `from xclim.core.calendar import parse_offset`